studies/american_gut/signals.py

```diff
@@ -1,21 +1,27 @@
-from django.db.models.signals import post_save
+import sys
+
+from django.db.models.signals import pre_save
 from django.dispatch import receiver
 
-from data_import.signal_helpers import task_signal
+from data_import.signal_helpers import task_signal_pre_save
 
 from .models import UserData, DataFile
 
 
-@receiver(post_save, sender=UserData)
-def post_save_cb(sender, instance, created, raw, update_fields, **kwargs):
+@receiver(pre_save, sender=UserData)
+def pre_save_cb(**kwargs):
     """
-    Initiate retrieval of the data corresponding to an American Gut survey ID.
+    Create data retrieval task when American Gut UserData's data is updated.
     """
-    if not instance.survey_ids:
+    # Skip this signal during tests.
+    if 'test' in sys.argv:
```

Comment:
I think we'll want to skip this test both for fixtures and on a per-test basis but not across the board so that we can actually test the signal itself. It's possible to silence signals with decorators, and the `if raw` test from the other signals should handle the case for fixtures.